core/src/main/scala/org/bitcoins/core/protocol/script/ScriptPubKey.scala

```diff
@@ -866,6 +866,70 @@ object NonStandardNotIfConditionalScriptPubKey
   }
 }
 
+sealed trait P2PKWithTimeoutScriptPubKey extends RawScriptPubKey {
+
+  val pubKey: ECPublicKey =
+    ECPublicKey.fromBytes(asm(2).bytes)
+
+  val lockTime: ScriptNumber = ScriptNumber.fromBytes(asm(5).bytes)
+
+  val timeoutPubKey: ECPublicKey =
+    ECPublicKey.fromBytes(asm(9).bytes)
+}
+
+object P2PKWithTimeoutScriptPubKey
+    extends ScriptFactory[P2PKWithTimeoutScriptPubKey] {
+  private case class P2PKWithTimeoutScriptPubKeyImpl(asm: Vector[ScriptToken])
+      extends P2PKWithTimeoutScriptPubKey
+
+  override def fromAsm(asm: Seq[ScriptToken]): P2PKWithTimeoutScriptPubKey = {
+    buildScript(
+      asm = asm.toVector,
+      constructor = P2PKWithTimeoutScriptPubKeyImpl.apply,
+      invariant = isP2PKWithTimeoutScriptPubKey,
+      errorMsg = s"Given asm was not a P2PKWithTimeoutScriptPubKey, got $asm"
+    )
+  }
+
+  def apply(
+      pubKey: ECPublicKey,
+      lockTime: ScriptNumber,
+      timeoutPubKey: ECPublicKey): P2PKWithTimeoutScriptPubKey = {
+    val timeoutAsm = CLTVScriptPubKey(lockTime, EmptyScriptPubKey).asm.toVector
```

Comment:
why `EmptyScriptPubKey` here? Can't we construct the nested `P2PK`/`P2PKH` ? The more I think about this, shouldn't `timeoutPubkey` really just be a generic `ScriptPubKey`?

Reply:
I think i understand what you are doing now, basically forcing both branches to be single public keys. Is there a reason both branches are forced to be public keys?

Reply:
I'm not trying to do something general of some kind of
```
OP_IF SPK_YIELDING_PUBKEY
OP_ELSE SPK_YIELDING_PUBKEY
OP_ENDIF
OP_CHECKSIG
```
I want a Pay to PubKey with a timeout pubkey option.
Should this pattern every show up again I'd likely be partial to adding something more general but I think it would be bad to try to do so with only one data point.